number_trivia_app/lib/features/number_trivia/presentation/pages/number_trivia_page.dart

```diff
@@ -18,52 +58,135 @@ class NumberTriviaPage extends StatelessWidget {
-                    return LoadingWidget();
-                  } else if (state is Loaded) {
-                    return TriviaDisplay(
-                      numberTrivia: state.trivia,
-                    );
-                  } else if (state is Error) {
-                    return MessageDisplay(
-                      message: state.message,
-                    );
-                  }
-                },
-              ),
-              SizedBox(
-                height: 20,
-              ),
-              TriviaControls(),
-            ],
+  Widget buildTrivia(NumberTriviaState state) {
+    if (state is Empty) {
+      return MessageDisplay(
+        message: 'Start searching',
+      );
+    } else if (state is Loading) {
+      return LoadingWidget();
+    } else if (state is Loaded) {
+      return TriviaDisplay(
+        numberTrivia: state.trivia,
+      );
+    } else {
+      return MessageDisplay(
+        message: state.message,
+      );
+    }
+  }
+
+  Widget buildTriviaControls(BuildContext context) {
+    return Column(
+      children: <Widget>[
+        TextField(
+          controller: controller,
+          keyboardType: TextInputType.number,
+          decoration: InputDecoration(
+            border: OutlineInputBorder(),
+            hintText: 'Input a number',
           ),
+          onChanged: (value) {
+            inputStr = value;
+          },
+          onSubmitted: (_) {
+            dispatchConcrete();
+          },
         ),
-      ),
+        SizedBox(
+          height: 10,
+        ),
+        Row(
+          children: <Widget>[
+            Expanded(
+              child: RaisedButton(
+                child: Text(
+                  'Search',
+                ),
+                color: Theme.of(context).accentColor,
+                textTheme: ButtonTextTheme.primary,
+                onPressed: dispatchConcrete,
+              ),
+            ),
+            SizedBox(
+              width: 10,
+            ),
+            Expanded(
+              child: RaisedButton(
+                child: Text(
+                  'Get random trivia',
+                ),
+                onPressed: dispatchRandom,
+              ),
+            ),
+          ],
+        )
+      ],
+    );
+  }
+
+  void dispatchConcrete() {
+    controller.clear();
+
+    final inputEither = inputConverter.stringToUnsignedInteger(inputStr);
+    inputEither.fold(
+      (failure) {
+        changeState(Error(message: INVALID_INPUT_FAILURE_MESSAGE));
+      },
+      (integer) async* {
+        changeState(Loading());
+        final failureOrTrivia = await getConcreteNumberTrivia(Params(number: integer));
+        yield* _eitherLoadedOrErrorState(failureOrTrivia);
+      },
     );
   }
+
+  Future<void> dispatchRandom() async {
+    controller.clear();
+
+    Loading();
+    final failureOrTrivia = await getRandomNumberTrivia(NoParams());
+    _eitherLoadedOrErrorState(failureOrTrivia);
```

Comment:
changeState()

Reply:
Seems you haven't pushed this change. Please consider doing push of intermediate commits in order to facilitate conversation about pending changes.